### ISSUE TITLE
Github-deployment-gate Integration 404s during initial setup

### ISSUE THREAD

**Issue by Fwang36** [getsentry/sentry#64566]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Go to [github-deployment-gate integration set up page](https://francisorg.sentry.io/settings/sentry-apps/github-deployment-gates/) in your org. 
2. Click on Accept & Install.
3. Click on Login with Github.
4. Receive 404 page

The url of the 404 has an undefined `client_id`.  Full url from my org is `https://github.com/login/oauth/authorize?client_id=undefined&scope=offline_access`

Trying to set up the integration from Github's side on the[ Github app's page](https://github.com/apps/sentry-deployment-gate) leads to the same result.

### Expected Result

Integration set up properly.

### Actual Result

Can not set up integration.

<img width="700" alt="Screenshot 2024-02-05 at 10 17 39 AM" src="https://github.com/getsentry/sentry/assets/96670625/2a1c641a-c0e2-4231-ac4f-5dcee67dd924">

### Product Area

Settings - Integrations

### Link

https://francisorg.sentry.io/settings/sentry-apps/github-deployment-gates/

**Comment by Dhrumil-Sentry** [comment]
Thanks for raising this issue, we're adding this to our backlog and will update here as we investigate

**Comment by d-haxton** [comment]
This is blocking us from using deployment gates and most of our integration with sentry.

**Comment by Dhrumil-Sentry** [comment]
@d-haxton while we investigate this, have you looked at this discussion: https://github.com/getsentry/sentry/discussions/55469#discussioncomment-7845224 ?

The Deployment gate integration only relies on new issues, but we're shipping something that lets you make deployment decisions based on metrics like error counts, new issue counts and crash free rates. You can easily call the API from your GH actions too.

**Comment by sentaur-athena** [comment]
@Fwang36 are you unblocked with the launch of new releases feature @Dhrumil-Sentry mentioned earlier?
